src/oic/oic/provider.py

```diff
@@ -1609,8 +1609,16 @@ def client_registration_setup(self, request):
 
         return response
 
-    def registration_endpoint(self, request, authn=None, **kwargs):
-        return self.l_registration_endpoint(request, authn, **kwargs)
+    def registration_endpoint(self, request, authn=None, method='POST', **kwargs):
+        if method.lower() == 'post':
+            return self.l_registration_endpoint(request, authn, **kwargs)
```

Comment:
The name of `l_registration_endpoint` is quite out of place now - feel like updating that?

Reply:
Was thinking about it :) Unfortunately it is not marked as private so someone might be using it.

Also the order of args is not consistent with the `read_registration` so if we decide to change it, I would change both name and order of args.

Reply:
OK, makes sense!